### PR TITLE
Legend colors and position

### PR DESCRIPTION
**Pull request by danielhkl**
tidied up pull request for legend colors and legend positions even outside of plot.
The last pull request was confusing in the end as well as my git history, so I started a new one.
Hopefully it is more obvious and understandable now.

### CODE REVIEW

**Comment by nschloe** [comment]
The PR still contains a wrong hash, and pylint complains.

**Comment by danielhkl** [comment]
@nschloe this is curious. On my local pc every test passes. I compared the output with my local output and it looks equal. So I didn't change anything in test_legends2 and from my point of view it works like it should. I don't know why the computed hash here differs from the old hash, especially because the old hash is equal to the one which is computed on my local pc.

**Comment by nschloe** [comment]
First of all, the lint error
```
pylint matplotlib2tikz/
************* Module matplotlib2tikz.legend
C:210, 0: Unnecessary parens after u'except' keyword (superfluous-parens)
```
should be easily fixed.

The hash difference may occur with different versions of any software involved; pgfplots, Pillow, etc. Upgrading your stack may help reproducing the hash locally.

If that doesn't work, you could simply replace the hash, ignoring your local failure, to make sure the travis tests pass.

File: matplotlib2tikz/legend.py [hunk 1]
```diff
@@ -193,6 +199,17 @@ def draw_legend(data, obj):
...
+    # Set color of lines in legend
+    data['legend colors'] = []
+    for handle in obj.legendHandles:
+        try:
+            data, legend_color, _ = mycol.mpl_color2xcolor(data,
+                                                            handle.get_color())
+            data['legend colors'].append('\\addlegendimage{no markers, %s}\n'
+                                        % legend_color)
+        except AttributeError:
+            data['legend colors'] = []
```
Comment (nschloe) [inline matplotlib2tikz/legend.py]: Can't we just `pass`?

File: matplotlib2tikz/legend.py [hunk 2]
```diff
@@ -193,6 +199,17 @@ def draw_legend(data, obj):
             'legend columns={}'.format(obj._ncol)
             )
 
+    # Set color of lines in legend
+    data['legend colors'] = []
```
Comment (nschloe) [inline matplotlib2tikz/legend.py]: This way, `'legend colors'` gets reset every for every new legend. Is this intended?

Comment (danielhkl) [inline matplotlib2tikz/legend.py]: You mean, it could be a problem in case you have more than one legend ?

File: matplotlib2tikz/legend.py (hunk 1 above)
Comment (danielhkl) [inline matplotlib2tikz/legend.py]: Yes, we can.

File: matplotlib2tikz/legend.py (hunk 2 above)
Comment (nschloe) [inline matplotlib2tikz/legend.py]: Indeed.

**Comment by nschloe** [comment]
This comes without an explicit test, but I'll merge it anyways. Hopefully I don't mess this up the future...

**Pull request merged**